modin/data_management/functions/groupby_function.py

```diff
@@ -101,8 +133,8 @@ def reduce(
         if not as_index:
             insert_levels = partition_idx == 0 and (drop or method == "size")
             result.reset_index(drop=not insert_levels, inplace=True)
-
-        return result
+        # Result could not always be a frame, so wrapping it into DataFrame
```

Comment:
Should we add check for this?

Reply:
check will be redundant, `pandas.DataFrame(obj)` converts `obj` to frame if `obj` is Series, and has no effect if `obj` is a DataFrame.

Reply:
ok